src/resourcesBuilder.js

```diff
@@ -0,0 +1,51 @@
+import {Delete, Resource} from 'admin-on-rest';
+import React from 'react';
+import Create from './Create';
+import Edit from './Edit';
+import List from './List';
+import Show from './Show';
+import defaultFieldFactory from './fieldFactory';
+import defaultInputFactory from './inputFactory';
+
+const makeResource = (resource, api, fieldFactory, inputFactory) => {
+  const {
+    create = Create,
+    edit = Edit,
+    list = List,
+    name,
+    props,
+    remove = Delete,
+    show = Show,
+  } = resource;
+  return (
+    <Resource
+      {...props}
+      create={create}
+      edit={edit}
+      key={name}
+      list={list}
+      name={name}
+      options={{
+        api,
+        fieldFactory,
+        inputFactory,
+        resource,
+      }}
+      remove={remove}
+      show={show}
+    />
+  );
+};
+
+/**
+ * Creates array of Resource components from Api instance 
+ * using provided input/field factories or default ones
+ */
+export default (
```

Comment:
This function should be in `AdminBuilder`:
* `AdminBuilder` already have the default value of `fieldFactory` and `inputFactory`.
* behaviour will be the same for all factories.